contracts/0.8.9/oracle/ValidatorsExitBusOracle.sol

```diff
@@ -0,0 +1,507 @@
+        bytes data;
+    }
+
+    /// @notice The list format of the validator exit requests data. Used when all
+    /// requests fit into a single transaction.
+    ///
+    /// Each validator exit request is described by the following 64-byte array:
+    ///
+    /// MSB <------------------------------------------------------- LSB
+    /// |  3 bytes   |  5 bytes   |     8 bytes      |    48 bytes     |
+    /// |  moduleId  |  nodeOpId  |  validatorIndex  | validatorPubkey |
+    ///
+    /// All requests are tightly packed into a byte array where requests follow
+    /// one another without any separator or padding, and passed to the `data`
+    /// field of the report structure.
+    ///
+    /// Requests must be sorted in the ascending order by the following compound
+    /// key: (moduleId, nodeOpId, validatorIndex).
+    ///
+    uint256 public constant DATA_FORMAT_LIST = 0;
+
+    /// @notice Submits report data for processing.
+    ///
+    /// @param data The data. See the `ReportData` structure's docs for details.
+    /// @param contractVersion Expected version of the oracle contract.
+    ///
+    /// Reverts if:
+    /// - The caller is not a member of the oracle committee and doesn't possess the
+    ///   SUBMIT_DATA_ROLE.
+    /// - The provided contract version is different from the current one.
+    /// - The provided consensus version is different from the expected one.
+    /// - The provided reference slot differs from the current consensus frame's one.
+    /// - The processing deadline for the current consensus frame is missed.
+    /// - The keccak256 hash of the ABI-encoded data is different from the last hash
+    ///   provided by the hash consensus contract.
+    /// - The provided data doesn't meet safety checks and boundaries.
+    ///
+    function submitReportData(ReportData calldata data, uint256 contractVersion) external {
+        _checkMsgSenderIsAllowedToSubmitData();
+        _checkContractVersion(contractVersion);
+        // it's a waste of gas to copy the whole calldata into mem but seems there's no way around
+        _checkConsensusData(data.refSlot, data.consensusVersion, keccak256(abi.encode(data)));
+        _startProcessing();
+        _handleConsensusReportData(data);
+    }
+
+    /// @notice Returns maximum number of validator exit requests that can be
+    /// submitted in the report for the current frame, accounting for both
+    /// the rate limit and the absolute limit per report.
+    ///
+    function getMaxExitRequestsForCurrentFrame() external view returns (uint256) {
+        return Math.min(
+            _storageDataBoundaries().value.maxRequestsPerReport,
+            RateLimit.load(RATE_LIMIT_POSITION).calculateLimitAt(_getCurrentRefSlot()) / 10**18
+        );
+    }
+
+    /// @notice Returns the total number of validator exit requests ever processed
+    /// across all received reports.
+    ///
+    function getTotalRequestsProcessed() external view returns (uint256) {
```

Comment:
why do we need store this? `TOTAL_REQUESTS_PROCESSED_POSITION`

Reply:
since exit requests can be brought in multiple transactions after Merkle data format is implemented

Reply:
I would rather leave at as is